smartsim/experiment.py

```diff
@@ -797,6 +798,51 @@ def reconnect_orchestrator(self, checkpoint: str) -> Orchestrator:
             logger.error(e)
             raise
 
+
+    def preview(
+        self,
+        *args: t.Any,
+        output_format: _OutputFormatString = None,
+        verbosity_level: _VerbosityLevelString = "info",
+        output_filename: t.Optional[str] = None,
+    ) -> None:
+        """Preview entity information prior to launch. This method
+        aggregates multiple pieces of information to give users insight
+        into what and how entities will be launched.  Any instance of
+        ``Model``, ``Ensemble``, or ``Orchestrator`` created by the
+        Experiment can be passed as an argument to the preview method.
+        :param output_format: Set output destination. The possible accepted
+        output formats are `json`, `xml`, `html`, `plain_text`, `color_text`.
+        A filename is required if an output format is specified. If no output
+        format is set, the preview will be output to stdout. Defaults to None.
+        :type output_type: str
+        :param output_filename: Specify name of path to write preview data to.
+        Only needed when an output format has been specified. Defaults to None.
+        :type output_filename: str
+        :param verbosity level: the verbosity level.
+        info: Display User defined fields and entities
+        debug: Display user defined field and entities and auto generated
+        fields.
+        developer: Display user defined field and entities, auto generated
+        fields, and run commands.
+        Defaults to info.
+        :type verbosity_level: str
+        """
+        if args:
+            preview_manifest = Manifest(*args)
+        else:
+            preview_manifest = None
+
+        rendered_preview = previewrenderer.render(
```

Comment:
are we sure that we want to mix different formattings in a single template? passing output_format (e.g. xml) would mean I'd create a single template that has all formats in a giant set of conditionals and it would potentially be very large.

Consider a separate template to keep the templates clean.  Maybe we could use format to select a subdir:

```
- <root>/templates/preview/{format}/base.tpl
|-----------------------------/exp.tpl
|-----------------------------/orch.tpl
- <root>/templates/preview/{format}/base.tpl
|-----------------------------/exp.tpl
|-----------------------------/orch.tpl
```

alternatively, file extensions...

```
- <root>/templates/preview/base_{format}.tpl
|-----------------------------/exp_txt.tpl
|-----------------------------/orch_txt.tpl
|-----------------------------/exp_xml.tpl
|-----------------------------/orch_xml.tpl
```

I'm not sure if a single file is better. Maybe @MattToast has an opinion?

Reply:
I went with the 2nd option for now